controllers/profiles/common/mutate_visitors.go

```diff
@@ -105,32 +105,28 @@ func ServiceMutateVisitor(workflow *operatorapi.SonataFlow) MutateVisitor {
 	}
 }
 
-func WorkflowPropertiesMutateVisitor(ctx context.Context, catalog discovery.ServiceCatalog,
-	workflow *operatorapi.SonataFlow, platform *operatorapi.SonataFlowPlatform) MutateVisitor {
+func UserPropertiesMutateVisitor(ctx context.Context, catalog discovery.ServiceCatalog,
+	workflow *operatorapi.SonataFlow, platform *operatorapi.SonataFlowPlatform, userProps *corev1.ConfigMap) MutateVisitor {
 	return func(object client.Object) controllerutil.MutateFn {
 		return func() error {
-			if kubeutil.IsObjectNew(object) {
-				return nil
-			}
-			cm := object.(*corev1.ConfigMap)
-			cm.Labels = workflow.GetLabels()
-			_, hasKey := cm.Data[workflowproj.ApplicationPropertiesFileName]
+			managedProps := object.(*corev1.ConfigMap)
+			managedProps.Labels = workflow.GetLabels()
+			_, hasKey := managedProps.Data[workflowproj.GetManagedPropertiesFileName(workflow)]
 			if !hasKey {
-				cm.Data = make(map[string]string, 1)
-				props, err := properties.ImmutableApplicationProperties(workflow, platform)
-				if err != nil {
-					return err
-				}
-				cm.Data[workflowproj.ApplicationPropertiesFileName] = props
-				return nil
+				managedProps.Data = make(map[string]string, 1)
+				managedProps.Data[workflowproj.GetManagedPropertiesFileName(workflow)] = ""
 			}
 
+			userProperties, hasKey := userProps.Data[workflowproj.ApplicationPropertiesFileName]
+			if !hasKey {
+				userProperties = ""
+			}
 			// In the future, if this needs change, instead we can receive an AppPropertyHandler in this mutator
 			props, err := properties.NewAppPropertyHandler(workflow, platform)
 			if err != nil {
 				return err
 			}
-			cm.Data[workflowproj.ApplicationPropertiesFileName] = props.WithUserProperties(cm.Data[workflowproj.ApplicationPropertiesFileName]).
+			managedProps.Data[workflowproj.GetManagedPropertiesFileName(workflow)] = props.WithUserProperties(userProperties).
```

Comment:
So the managed properties CM will have also user properties? Did I understand this correctly? Do we need both properties in the managed CM?

Reply:
no, the `userProps` are passed on to the property builder so that it can generate the discovery properties that, instead, are part of the managed properties.
see line 99 of `application.go`: `userProps` are not passed to the builder of the `managedProps`

Reply:
OH YEAH, you're right! Thanks for the clarification!

Reply:
I think this code for people that comes from the previous version is a bit confusing in the first read, because before, the user properties where combined with the properties we where generating. And now, with the internal refactoring you did, the user properties are  being used as a kind of "candidates for doing the service discovery", but won't be included in the produced properties set.

Maybe, now that we are going to keep to properties living in parallel, the original application.properties, plus the new application-pod.properties generated by this handler, we can rename to something like ManagedAppPropertiesHandler (or a better name)  to distinguish the new situation.

Just thinking aloud.